widget/embedded/src/hooks/useSwapInput.ts

```diff
@@ -113,15 +122,29 @@ export function useSwapInput(): UseSwapInput {
       fetchQuote(requestBody)
         .then((res) => {
           setLoading(false);
-          setQuote(res);
-          throwErrorIfResponseIsNotValid(res);
-          const quoteWarning = generateQuoteWarnings(res, {
-            fromToken,
-            toToken,
-            userSlippage,
-            tokens,
-          });
+          setQuotes(res);
+          const sortQuotes = sortRoutesBy(sortStrategy, res.results);
+          const quote = sortQuotes.length ? sortQuotes[0] : null;
+          const sortedQuoteTags = sortTags(quote?.tags || []);
+          setSelectedQuote(quote);
+          setSelectedTag(sortedQuoteTags[0]);
+          const { requestAmount } = res;
+
+          const quoteWarning =
+            quote &&
+            generateQuoteWarnings(quote, {
+              fromToken,
+              toToken,
+              userSlippage,
+              tokens,
+              requestAmount,
+            });
           setWarning(quoteWarning);
+          throwErrorIfResponseIsNotValid({
+            diagnosisMessages: res.diagnosisMessages,
+            requestId: quote?.requestId || '',
+            swaps: quote?.swaps,
+          });
```

Comment:
I think it would be better to move this code to the top, just like before. Validating the response first and then proceeding with additional logic is a more effective approach.

Reply:
fixed